pkg/parser/whitelist_test.go

```diff
@@ -0,0 +1,302 @@
+package parser
+
+import (
+	"testing"
+
+	"github.com/crowdsecurity/crowdsec/pkg/models"
+	"github.com/crowdsecurity/crowdsec/pkg/types"
+	log "github.com/sirupsen/logrus"
+)
+
+func TestWhitelistCompile(t *testing.T) {
+	node := &Node{
+		Logger: log.NewEntry(log.New()),
+	}
+	tests := []struct {
+		name         string
+		whitelist    Whitelist
+		expected_err bool
+	}{
+		{
+			name: "Valid CIDR whitelist",
+			whitelist: Whitelist{
+				Reason: "test",
+				Cidrs: []string{
+					"127.0.0.1/24",
+				},
+			},
+		},
+		{
+			name: "Invalid CIDR whitelist",
+			whitelist: Whitelist{
+				Reason: "test",
+				Cidrs: []string{
+					"127.0.0.1/1000",
+				},
+			},
+			expected_err: true,
+		},
+		{
+			name: "Valid EXPR whitelist",
+			whitelist: Whitelist{
+				Reason: "test",
+				Exprs: []string{
+					"1==1",
+				},
+			},
+		},
+		{
+			name: "Invalid EXPR whitelist",
+			whitelist: Whitelist{
+				Reason: "test",
+				Exprs: []string{
+					"evt.THISPROPERTYSHOULDERROR == true",
+				},
+			},
+			expected_err: true,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			node.Whitelist = tt.whitelist
+			_, err := node.CompileWLs()
+			if err == nil && tt.expected_err {
```

Comment:
you can test for the actual error message, see how we use

cstest.RequireErrorContains(t, err, tc.expected_err)